### PR TITLE
Fix broken padding docs anchor in conv and pool docstrings

### LINKED ISSUE #91613: Notes on padding missing in documentation
### Issue type

Documentation Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf 2.16.1 

### Custom code

No

### OS platform and distribution

_No response_

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

The link for the `padding` argument in the [conv1d_transpose](https://www.tensorflow.org/api_docs/python/tf/nn/conv1d_transpose) documentation currently redirects to the top of the [tf.nn documentation page](https://www.tensorflow.org/api_docs/python/tf/nn), instead of the relevant section.

### Expected Behavior

The [link](https://www.tensorflow.org/api_docs/python/tf/nn#notes_on_padding_2) should redirect to the appropriate section that documents the `padding` argument, ideally pointing to the relevant source code for context:

https://github.com/tensorflow/tensorflow/blob/4dbae670e3a5c6e15c2dd9192ee229e56c84a3c7/tensorflow/python/ops/nn_ops.py#L17-L166

### Standalone code to reproduce the issue

```shell
Not applicable.
```

### Relevant log output

```shell

```

### PR DESCRIPTION
**Pull request by VibhorGautam**
The padding parameter in every conv and pool function links to tf.nn#notes_on_padding_2, but that anchor does not exist on the rendered API docs page. Clicking the link just scrolls to the top of the tf.nn module page instead of the "Notes on padding" section.

Updated all 24 occurrences across nn_ops.py (23) and nn_impl.py (1) to use the correct anchor fragment without the stale _2 suffix.

Fixes #91613

### CODE REVIEW

**Review by ratgr — APPROVED** [review]
Thank you for this contribution! the changes look good
